Shared/Samples/Generate offline map with custom parameters/GenerateOfflineMapWithCustomParametersView.Model.swift

```diff
@@ -0,0 +1,308 @@
+            return try await offlineMapTask.makeDefaultGenerateOfflineMapParameters(
+                areaOfInterest: areaOfInterest
+            )
+        }
+        
+        /// Creates the generate offline map parameter overrides.
+        /// - Parameter parameters: The generate offline map parameters.
+        /// - Returns: A `GenerateOfflineMapParameterOverrides` if there are no errors.
+        private func makeGenerateParameterOverrides(
+            parameters: GenerateOfflineMapParameters
+        ) async throws -> GenerateOfflineMapParameterOverrides {
+            // Returns the overrides.
+            return try await offlineMapTask.makeGenerateOfflineMapParameterOverrides(
+                parameters: parameters
+            )
+        }
+        
+        /// Sets up the model by getting the generate offline map parameters and parameter
+        /// overrides.
+        /// - Parameter areaOfInterest: The area of interest to create the parameters and parameter
+        /// overrides for.
+        func setUpParametersAndOverrides(extent: Envelope) async throws {
+            offlineMapParameters = try await makeGenerateOfflineMapParameters(areaOfInterest: extent)
+            if let offlineMapParameters {
+                offlineMapParameterOverrides = try await makeGenerateParameterOverrides(
+                    parameters: offlineMapParameters
+                )
+            }
+        }
+        
+        /// Generates the offline map.
+        func generateOfflineMap() async throws {
+            // Disables the generate offline map button.
+            isGenerateDisabled = true
+            
+            guard let parameters = offlineMapParameters,
+                  let overrides = offlineMapParameterOverrides,
+                  let extent = parameters.areaOfInterest?.extent else { return }
+            
+            // Creates the generate offline map job based on the parameters and overrides.
+            generateOfflineMapJob = offlineMapTask.makeGenerateOfflineMapJob(
+                parameters: parameters,
+                downloadDirectory: temporaryDirectory,
+                overrides: overrides
+            )
+            
+            // Starts the job.
+            generateOfflineMapJob.start()
+            
+            defer {
+                generateOfflineMapJob = nil
+                isGenerateDisabled = offlineMap != nil
+            }
+            
+            // Awaits the output of the job.
+            let output = try await generateOfflineMapJob.output
+            // Sets the offline map to the output's offline map.
+            offlineMap = output.offlineMap
+            // Sets the initial viewpoint of the offline map.
+            offlineMap.initialViewpoint = Viewpoint(boundingGeometry: extent.expanded(by: 0.8))
+        }
+        
+        /// Cancels the generate offline map job.
+        func cancelJob() async {
+            // Cancels the generate offline map job.
```

Comment:
```suggestion
        /// Cancels the generate offline map job.
        func cancelJob() async {
```

This comment is the same as the doc comment.